### ISSUE TITLE
ReconnectInterval not picked from [SESSION] sections, only from [DEFAULT] section

### ISSUE THREAD

**Issue by cebaa** [quickfix/quickfix#604]
From "Sample Settings File" section of the [docs](https://quickfixengine.org/c/documentation/), this configuration:

```
[DEFAULT]
ReconnectInterval=60

[SESSION]
# overide default setting for RecconnectInterval
ReconnectInterval=30
```

should make the session have reconnect interval of 30 instead of the default of 60. However, that doesn't seem to be working. Looking at the code for [SocketInitiator](https://github.com/quickfix/quickfix/blob/94625580bfd50c1ebba3f704a88c9e3c392f9fb1/src/C%2B%2B/SocketInitiator.cpp#L70)

```
  const Dictionary& dict = s.get();
```

and the code for `get` in {SessionSettings](https://github.com/quickfix/quickfix/blob/94625580bfd50c1ebba3f704a88c9e3c392f9fb1/src/C%2B%2B/SessionSettings.h#L259):

```
  /// Get global default settings
  const Dictionary& get() const { return m_defaults; }
```

it looks to me this is only picking up the default value. 

In other words, all sessions must have the same reconnect interval, either [the default 30](https://github.com/quickfix/quickfix/blob/94625580bfd50c1ebba3f704a88c9e3c392f9fb1/src/C%2B%2B/SocketInitiator.cpp#L38) or whatever is defined in `[DEFAULT]` section.

**Comment by pyy07** [comment]
i found this too

**Comment by cuongth** [comment]
We can fix it in SocketInitiator::doConnect
https://github.com/quickfix/quickfix/blob/94625580bfd50c1ebba3f704a88c9e3c392f9fb1/src/C%2B%2B/SocketInitiator.cpp#L131
we add following
```
if( d.has( RECONNECT_INTERVAL ) ) {
      m_reconnectInterval = d.getInt( RECONNECT_INTERVAL );
}
```
It will overwrite assignment in SocketInitiator::onConfigure. I will update it in my 'linux' branch tomorrow.

**Comment by cuongth** [comment]
Kindly prefer my branch for the solution: ReconnectInterval value in [SESSION] dominates the value in [DEFAULT].

[ReconnectInterval value in [SESSION] dominates the value in [DEFAULT]](https://github.com/cuongth/quickfix/blob/linux/src/C%2B%2B/SocketInitiator.cpp#L142)

**Comment by cuongth** [comment]
> Kindly prefer my branch for the solution:
> 
> [ReconnectInterval value in [SESSION] dominates the value in [DEFAULT]](https://github.com/cuongth/quickfix/blob/linux/src/C%2B%2B/SocketInitiator.cpp#L142)
> 
> I also created a pull request #609 , you can prefer the solution there

Done. My branch is merged into master.

**Comment by orenmnero** [comment]
Fixed by PR #609 (merged 2024-12-18). `SocketInitiator::doConnect` now reads `ReconnectInterval` from the per-session dictionary, allowing `[SESSION]` values to override `[DEFAULT]`.
